### ISSUE TITLE
Support for templates in configuration files on file backend

### ISSUE THREAD

**Issue by hpello** [containous/traefik#2044]
### Do you want to request a *feature* or report a *bug*?
Bug

### What did you do?

The [documentation](https://docs.traefik.io/configuration/backends/file/) suggests that go templates or sprig functions may be used for file configuration backends.

I tried to define a configuration using go template in a TOML file.
```
cat traefik.toml
[file]
directory = "conf/"
```

```
cat conf/sub.toml
[backends]
  [backends.backend1]
  url = "http://firstserver"
  [backends.backend2]
  url = "http://secondserver"

{{$frontends := dict "frontend1" "backend1" "frontend2" "backend2"}}
[frontends]
{{range $frontend, $backend := $frontends}}
  [frontends.{{$frontend}}]
  backend = "{{$backend}}"
{{end}}
```

### What did you expect to see?
No error

### What did you see instead?
An error : `Error starting provider *file.Provider: error reading configuration file: Near line 5 (last key parsed 'backends.backend2'): bare keys cannot contain '{'`

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.0-rc1
Codename:     roquefort
Go version:   go1.9
Built:        2017-08-29_08:16:35AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```json
{
 "GraceTimeOut": 10000000000,
 "Debug": false,
 "CheckNewVersion": true,
 "AccessLogsFile": "",
 "AccessLog": null,
 "TraefikLogsFile": "",
 "LogLevel": "ERROR",
 "EntryPoints": {},
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": {
  "ReadTimeout": 0,
  "WriteTimeout": 0,
  "IdleTimeout": 180000000000
 },
 "ForwardingTimeouts": {
  "DialTimeout": 30000000000,
  "ResponseHeaderTimeout": 0
 },
 "Docker": null,
 "File": null,
 "Web": null,
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ConfigFile": ""
}
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
INFO[0000] Using TOML configuration file /home/ubuntu/deploy-swarm/sample.toml
INFO[2017-09-01T11:50:37Z] Traefik version v1.4.0-rc1 built on 2017-08-29_08:16:35AM
DEBU[2017-09-01T11:50:37Z] Global configuration loaded {"GraceTimeOut":10000000000,"Debug":true,"CheckNewVersion":true,"AccessLogsFile":"","AccessLog":null,"TraefikLogsFile":"","LogLevel":"DEBUG","EntryPoints":{"http":{"Network":"","Address":":80","TLS":null,"Redirect":null,"Auth":null,"WhitelistSourceRange":null,"Compress":false,"ProxyProtocol":false}},"Cluster":null,"Constraints":[],"ACME":null,"DefaultEntryPoints":["http"],"ProvidersThrottleDuration":2000000000,"MaxIdleConnsPerHost":200,"IdleTimeout":0,"InsecureSkipVerify":false,"RootCAs":null,"Retry":null,"HealthCheck":{"Interval":30000000000},"RespondingTimeouts":{"ReadTimeout":0,"WriteTimeout":0,"IdleTimeout":180000000000},"ForwardingTimeouts":{"DialTimeout":30000000000,"ResponseHeaderTimeout":0},"Docker":null,"File":{"Watch":true,"Filename":"/home/ubuntu/deploy-swarm/sample.toml","Constraints":null,"Trace":false,"DebugLogGeneratedTemplate":false,"Directory":"/conf/"},"Web":null,"Marathon":null,"Consul":null,"ConsulCatalog":null,"Etcd":null,"Zookeeper":null,"Boltdb":null,"Kubernetes":null,"Mesos":null,"Eureka":null,"ECS":null,"Rancher":null,"DynamoDB":null}
INFO[2017-09-01T11:50:37Z] Preparing server http &{Network: Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] Compress:false ProxyProtocol:false} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s
INFO[2017-09-01T11:50:37Z] Starting provider *file.Provider {"Watch":true,"Filename":"/home/ubuntu/deploy-swarm/sample.toml","Constraints":null,"Trace":false,"DebugLogGeneratedTemplate":false,"Directory":"/conf/"}
INFO[2017-09-01T11:50:37Z] Starting server on :80
ERRO[2017-09-01T11:50:37Z] Error starting provider *file.Provider: error reading configuration file: Near line 5 (last key parsed 'backends.backend2'): bare keys cannot contain '{'
```

**Comment by ldez** [comment]
Sorry the documentation is wrong, you cannot use template with the File Backend.

**Comment by hpello** [comment]
Ok, good to know. That's too bad, this would be a killer feature for a less verbose declaration of static frontends and backends!

I am using traefik in a micro-service architecture with Docker, and issue #1689 currently is a blocker for my deployment (dynamic frontends are not persisted, and 404 error codes are returned, which is not consistent with my api which expects 50x).

Just so you know, as a workaround I ended up dropping docker labels and statically declaring my backends/frontends in an custom _docker_ template file... as they support go/sprig templates features.

**Comment by timoreimann** [comment]
Solving #1689 for Docker seems doable. Let me have a look tonight.

**Comment by traefiker** [comment]
Closed by #2048.
